software/controller/lib/debugvars/vars.h

```diff
@@ -137,4 +137,40 @@ class FloatArray : public Base {
   std::array<float, N> data;
 };
 
+template <size_t N>
+class String : public Base {
+ public:
+  String(const char *name, Access access, const char *help = "", const char *fmt = "%s",
+         const char *units = "")
+      : Base(Type::String, name, access, units, help, fmt) {}
+
+  String(const char *name, Access access, const char *initial, size_t size, const char *help = "",
+         const char *fmt = "%s", const char *units = "")
+      : Base(Type::String, name, access, units, help, fmt) {
+    set(initial, size);
+  }
+
+  void get_value(void *write_buff) override { std::memcpy(write_buff, data.data(), size()); }
+
+  void set_value(void *read_buf) override { std::memcpy(data.data(), read_buf, size()); }
+
+  size_t size() const override { return N; }
+
+  void set(const char *new_value, size_t size) {
+    size = std::min(size, N);
+    std::memcpy(data.data(), new_value, size);
+  }
+
+  const char *get() const { return data.data(); }
+
+  std::array<char, N> data = {0};
```

Comment:
Strict googlestyle would be `data_`

Reply:
As in previous PR, this is ok if we interpret the style guide to mean public vs. non-public members.

Reply:
OK, but why should `data` be public?